proto/redpanda/api/dataplane/v1alpha2/pipeline.proto

```diff
@@ -378,3 +396,16 @@ message StartPipelineRequest {
 message StartPipelineResponse {
   Pipeline pipeline = 1;
 }
+
+message GetPipelineServiceConfigSchemaRequest {
+  enum Format {
+    FORMAT_UNSPECIFIED = 0;
+    FORMAT_JSONSCHEMA = 1;
+  }
+
+  Format format = 1;
```

Comment:
Do we anticipate multiple formats in the future? Should we omit until it's needed (In true [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) style?)

Reply:
I have added more format enums and have updated description with additional information regarding the formats.

Reply:
Is there a use case for each of these? It's going to me more work in the dataplane to add support for each of these formats.

Reply:
We want at least the JSON Schema one so we can generate it for frontend integration. I think one of the JSON ones (probably full or full scrubbed) would also be useful in case we ever want to have backend linting or validation within Console. Basically I added all for completeness to match Redpanda Connect. But for right now we really only need JSON Schema. I am open to just basically having unspecified (this is kinda required according to best practices and buf linting) and JSON Schema, and by default it would just return JSON Schema. If we need more later, we can add formats later.

Reply:
I agree with keeping it simple until we really need a new format.
If right now we only need JSON Schema, then let's remove the enum.
If/When we need another format, the enum can be easily added.